text/0000-allow-sharing-of-LWC-across-namespaces.md

```diff
@@ -0,0 +1,230 @@
+    <isExposed>true</isExposed>
+</LightningComponentBundle>
+```
+
+```
+            <!-- NamespaceBundle.xml -->
+<?xml version="1.0" encoding="UTF-8"?>
+<NamespaceBundle xmlns="http://soap.sforce.com/2006/04/metadata">
+    <namespaces>
+        <namespace>*</namespace> <!-- open to the world -->
+        
+        <!-- open to following namespaces
+            <namespace>force</namespace>
+            <namespace>forcechatter</namespace>
+            <namespace>ui</namespace>
+        -->
+    </namespaces>
+</NamespaceBundle>
+```
+
+##### Pros
+- Leverages isExposed tag without introducing any new tag.
+- Sharing is easy as you have to edit one single file to add or remove access to the namespaces.
+
+##### Cons
+- Individual component cannot be expose to different namespaces. It is controlled at namespace level so either all or none. 
+- Learning curve as we are extending isExposed flag.
+
+
+**Alernative #2** - Granular expose control 
+Add <exposed> tag to existing LightingComponentBundle file. This tag can contain list of namespaces where this components can be accessible. 
+
+
+__Note__ : isExposed flag has higher precedence over shared flag. 
+
+|  isExposed 	|   shared   	        |   Comments  	                |
+|---	        |---	                |---	                        |
+|  True 	    |  (ignored)	        | Open to the world  	        |
+|  True 	    |  (ignored)	        | Open to the world  	        |
+|  False     	|  ! isEmpty             | Open to Limited Namespaces  	|
+|  False     	|  isEmpty	            | Open to Local Namespace  	    |
+
+```
+        <!-- LightingComponentBundle.xml -->
+<?xml version="1.0" encoding="UTF-8"?>
+<LightningComponentBundle xmlns="http://soap.sforce.com/2006/04/metadata">
+    <apiVersion>49.0</apiVersion>
+    <isExposed>false</isExposed>
+    <shared>
+        <namespaces>
+            <namespace>force</namespace>
+            <namespace>forcechatter</namespace>
+            <namespace>ui</namespace>
+        </namespaces>
+    </shared>
+</LightningComponentBundle>
+```
+
+##### Pros
+- No need to introduce new xml file at namespace level since we are using existing LightingComponentBundle file.
+- Honor isExposed flag.
+- Granular control as we can control sharing of individual LWC. 
+
+##### Cons
+- Changing access for all namespace components involves changing all LWC bundle files. Too much of work for the developer.
+- Learning curve as we are introducing new flag.
+- Complex implementation as we have to set precedence logic in the code. 
+
+## Drawbacks
+
+- implementation cost, both in term of code size and complexity
+- whether the proposed feature can be implemented in user space
+- the impact on teaching people Lightning Web Components
+- integration of this feature with other existing and planned features
+- cost of migrating existing Lightning Web Components applications (is it a breaking change?)
+
+There are tradeoffs to choosing any path. Attempt to identify them here.
+

```

Comment:
Alternative 3... Fine grain control at the component level:

<LightningComponentBundle xmlns="http://soap.sforce.com/2006/04/metadata">
    <apiVersion>49.0</apiVersion>
    <isExposed>false</isExposed>
    <exposedTo>
        <namespace>force</namespace>
        <namespace>forcechatter</namespace>
        <namespace>ui</namespace>
    </exposedTo>
</LightningComponentBundle>

With the option of broader control at the namespace level:

<NamespaceBundle xmlns="http://soap.sforce.com/2006/04/metadata">
    <exposedTo>
        <namespace>force</namespace>
        <namespace>forcechatter</namespace>
        <namespace>ui</namespace>
    </exposedTo>
</NamespaceBundle>

Each component decide whether or not it `isExposed`. Each component has the option to declare who it is `exposedTo`. If it's exposed and doesn't limit who it's exposedTo, then it's exposed to all the namespaces specified in the NamespaceBundle. Essentially your component bundle has precedence.

Without fine grain control, you're going to hit the issue of over exposing. For example, I have a new component and I want it to be exposed to namespace "Foo". My only option is to modify the NamespaceBundle to include "Foo". Darn, I also have to expose all other components in this namespace to "Foo".

Reply:
The use case you mention is it valid are our customer doing that now? As per internal discussion, we decided that components within the namespaces are either all exposed to not. We don't have a case of cherry picking here. Please let us know if that is the case.